server/vulnerabilities/nvd/cve.go

```diff
@@ -147,20 +147,24 @@ func TranslateCPEToCVE(
 
 	var newVulns []fleet.SoftwareVulnerability
 	for _, vuln := range vulns {
-		newCount, err := ds.InsertSoftwareVulnerabilities(ctx, []fleet.SoftwareVulnerability{vuln}, fleet.NVDSource)
+		ok, err := ds.InsertSoftwareVulnerability(ctx, vuln, fleet.NVDSource)
 		if err != nil {
 			level.Error(logger).Log("cpe processing", "error", "err", err)
 			continue
 		}
 
-		// collect vuln only if newCount > 0, otherwise we would send
+		// collect vuln only if inserted, otherwise we would send
 		// webhook requests for the same vulnerability over and over again until
 		// it is older than 2 days.
-		if collectVulns && newCount > 0 {
+		if collectVulns && ok {
 			newVulns = append(newVulns, vuln)
 		}
 	}
 
+	if err = ds.DeleteOutOfDateVulnerabilities(ctx, fleet.NVDSource, 2*time.Hour); err != nil {
```

Comment:
Why 2 hours? Is this assuming vulnerability periodicity is 1 hour? (Or is it just to be safe assuming a terrible case scenario where the above loop takes 2 hours to execute?)

Reply:
Thanks for bringing this up - it was the latter, but on second thought I think that maybe doing some thing like this makes some sense? https://github.com/fleetdm/fleet/pull/10514/commits/da938427527e0fe760d9d668a3b1b5a5f5c39f91